runtime-config-linux.md

```diff
@@ -0,0 +1,121 @@
+If a path is specified, that particular file is used to join that type of namespace.
+
+```json
+    "namespaces": [
+        {
+            "type": "pid",
+            "path": "/proc/1234/ns/pid"
+        },
+        {
+            "type": "net",
+            "path": "/var/run/netns/neta"
+        },
+        {
+            "type": "mnt",
+        },
+        {
+            "type": "ipc",
+        },
+        {
+            "type": "uts",
+        },
+        {
+            "type": "user",
+        },
+    ]
+```
+
+#### Namespace types
+
+* **pid** processes inside the container will only be able to see other processes inside the same container.
+* **network** the container will have it's own network stack.
+* **mnt** the container will have an isolated mount table.
+* **ipc** processes inside the container will only be able to communicate to other processes inside the same
+container via system level IPC.
+* **uts** the container will be able to have it's own hostname and domain name.
+* **user** the container will be able to remap user and group IDs from the host to local users and groups
+within the container.
+
+### Access to devices
+
+Devices is an array specifying the list of devices from the host to make available in the container.
+By providing a device name within the list the runtime should look up the same device on the host's `/dev`
+and collect information about the device node so that it can be recreated for the container.  The runtime
+should not only create the device inside the container but ensure that the root user inside 
+the container has access rights for the device.
+
+```json
+   "devices": [
+        "null",
+        "random",
+        "full",
+        "tty",
+        "zero",
+        "urandom"
+    ]
+```
+
+## Linux control groups
+
+Also known as cgroups, they are used to restrict resource usage for a container and handle
+device access.  cgroups provide controls to restrict cpu, memory, IO, and network for
+the container. For more information, see the [kernel cgroups documentation](https://www.kernel.org/doc/Documentation/cgroups/cgroups.txt)
+
+## Linux sysctl
+
+sysctl allows kernel parameters to be modified at runtime for the container.
+For more information, see [the man page](http://man7.org/linux/man-pages/man8/sysctl.8.html)
+
+```json
+   "sysctl": {
+        "net.ipv4.ip_forward": "1",
+        "net.core.somaxconn": "256"
+   }
+```
+
+## Linux rlimits
+
+```json
+   "rlimits": [
+        {
+            "type": "RLIMIT_NPROC",
+            "soft": 1024,
+            "hard": 102400
+        }
+   ]
+```
+
+rlimits allow setting resource limits. The type is from the values defined in [the man page](http://man7.org/linux/man-pages/man2/setrlimit.2.html). The kernel enforces the soft limit for a resource while the hard limit acts as a ceiling for that value that could be set by an unprivileged process.
+
+## Linux user namespace mappings
```

Comment:
@philips What I was trying to say was that one could use the same mappings on any host and they would work assuming that the linux host has user namespaces enabled. I think you are referring to assigning a particular range to a container/cluster user and that may or may not be available on a host and hence isn't portable. One way to get around it is to just have users specify that they need a range of a particular length and then translate that to whatever is available on the host. However, that seems like a feature that belongs to higher level orchestration and it could be translated to non portable section of the OCI spec.